### ISSUE TITLE
MemorySnapshotStore should serialize the snapshot object received in SaveSnapshot.

### ISSUE THREAD

**Issue by basbossinkdivverence** [akkadotnet/akka.net#3157]
Akka.Net version: 1.3.1
Platform: windows.

When looking at the implementation of `SaveAsync` of `MemorySnapshotStore` it can be seen that it just stores the given object reference in a `List` this means that if the object is mutable and updated by other parts of the code the meaning of snapshot becomes a bit strange. Since when using a _real_ database by definition the snapshot gets serialized I _feel_ it is part of the contract of a `SnapshotStore`.

Given the current behavior it would be necessary to either implement a _copy_ of the snapshot which is actually only needed during testing, furthermore creating a _copy_ waists resources unnecessarily. Or use a file backed database during testing which bring a large set of problems with it. Or make the production code _aware_ of the distinction between _production_ and _testing_.

**Comment by Aaronontheweb** [comment]
@basbossinkdivverence agree, it should serialize a copy of it to give it immutable semantics.
